flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaProducerTestBase.java

```diff
@@ -255,7 +255,7 @@ protected void testOneToOneAtLeastOnce(boolean regularSink) throws Exception {
 
```

Comment:
FLINK-14235 was most likely caused by the bad test isolation of testOneToOneAtLeastOnce.

kafkaServer.unblockProxyTraffic() is not called once this test fails with `fail("Job should fail!");`

Reply:
FLINK-14235 was another test case.
This test failed because of the race condition of snapshot and EOI.
The other test failed because if this test executed first and failed, proxy traffic was still blocked (so no data ever arrived at Kafka). Hence, the successfully written elements are empty ([] in actual).